app/main/views/services.py

```diff
@@ -432,6 +433,7 @@ def update_section_submission(service_id, section_id):
     else:
         update_data.update(uploaded_documents)
 
+    if not errors and has_changes_to_save(section, draft, update_data):
```

Comment:
This is fine as long as we have document uploads on their own page (as we do), but if there were pages with a mix of questions and document uploads (as we had last time) then any validation errors on the questions wouldn't make it to the page, only the document errors.

This probably isn't a problem, because we'll probably stick with document uploads on their own page.